src/App.tsx

```diff
@@ -15,71 +15,109 @@ export const goods = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <main className="section container">
-    <h1 className="title">No goods selected</h1>
-
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
-
-      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
-
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
-
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
-
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Garlic
-          </td>
-        </tr>
-      </tbody>
-    </table>
-  </main>
-);
+type State = {
+  selectedGoods: string,
+};
+
+function removeClass(
+  collection: NodeListOf<HTMLButtonElement | HTMLTableRowElement>,
+  className: string,
+) {
+  collection.forEach(element => {
+    element.classList.remove(className);
+  });
+}
+
+export class App extends Component<{}, State> {
+  state: Readonly<State> = {
+    selectedGoods: 'Jam',
+  };
+
+  unsetCells = () => {
+    const cells = document.querySelectorAll('tr');
+    const buttons = document.querySelectorAll('button');
+
+    buttons.forEach((b: HTMLButtonElement) => {
+      // eslint-disable-next-line no-param-reassign
+      b.innerText = '+';
+    });
+
+    removeClass(buttons, 'is-info');
+    removeClass(cells, 'has-background-success-light');
```

Comment:
no need to work with DOM.
Work with state and decide which classes add to remove, which text set in button and so on conditionally in JSX.